src/components/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,176 @@
+    onChange?: (value?: string) => void;
+    onEnterPressed?: (value?: string) => void;
+    onKeyDown?: (event: KeyboardEvent<HTMLTextAreaElement>) => void;
+} & Omit<InputSharedBaseProps, 'hugWidth'> &
+    Omit<TextareaHTMLAttributes<HTMLTextAreaElement>, 'onChange'> &
+    Omit<TextareaAutosizeProps, 'onChange'>;
+
+export const Textarea = ({
+    autocomplete,
+    autosize = false,
+    clearable = false,
+    decorator,
+    debounceTime = 500,
+    defaultValue,
+    disabled = false,
+    extraActions = undefined,
+    focusOnMount = false,
+    id: propId,
+    minRows,
+    maxRows,
+    placeholder,
+    readOnly,
+    resizable = false,
+    required,
+    selectable = false,
+    status = Validation.Default,
+    value,
+    onBlur,
+    onChange,
+    onEnterPressed,
+    onKeyDown,
+    onFocus,
+    'data-test-id': dataTestId = 'fondue-textarea',
+    ...props
+}: TextareaProps): ReactElement => {
+    const Component = autosize ? TextareaAutosize : 'textarea';
+
+    const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+
+    const { isFocusVisible, focusProps } = useFocusRing({
+        isTextInput: true,
+        within: true,
+    });
+
+    const handleOnChange = useDebounce(() => {
+        if (textareaRef.current) {
+            const { value } = textareaRef.current;
+            onChange?.(value);
+        }
+    }, debounceTime);
+
+    const handleOnKeyDown = (event: KeyboardEvent<HTMLTextAreaElement>) => {
+        if (onEnterPressed && event.key === 'Enter') {
+            onEnterPressed(textareaRef.current?.value);
+        } else {
+            onKeyDown?.(event);
+        }
+    };
+
+    const handleClear = () => {
+        if (textareaRef.current) {
+            textareaRef.current.value = '';
+        }
+    };
+
+    useEffect(() => {
+        focusOnMount && textareaRef.current?.focus();
+    }, [focusOnMount]);
+
+    const autosizeProps = { minRows, maxRows };
+
+    const getPaddingRight = () => {
+        let paddingClass;
+        switch (true) {
+            case clearable && status !== Validation.Default:
+                paddingClass = 'tw-pr-[3.5rem]';
+                break;
+            case status !== Validation.Default || clearable:
+                paddingClass = 'tw-pr-[2.5rem]';
+                break;
+            default:
+                paddingClass = 'tw-pr-[1rem]';
+        }
+        return paddingClass;
+    };
+
+    return (
+        <div className={'tw-relative'}>
+            {decorator ? (
+                <div
+                    className={merge([
+                        'tw-absolute tw-top-[0.7rem] tw-left-[0.7rem] tw-z-10',
+                        disabled ? InputStyles.disabled : '',
+                        readOnly ? InputStyles.readOnly : '',
+                    ])}
+                    data-test-id={`${dataTestId}-decorator`}
+                >
+                    {decorator}
+                </div>
+            ) : null}
+            <span {...focusProps}>
```

Comment:
`span` element are usually not focusable. Would that be correct from an accessibility point of view?

Reply:
The reason for this it there is a overwrite conflict with the `style` prop on the `textarea` when `react-aria` is applied directly.  So the `textarea` is wrapped in a `span` and the `useFocuseRing` hook is set to `within` to trigger when the `textarea` is focused within the `span`.  This has to be the structure as we also have faceable buttons `within` the same parent to the `textarea` needs its own wrapper.  Not sure of another approach here.